modules/cms-kit/src/Volo.CmsKit.Public.Web/DefaultHomePageMiddleware.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Volo.CmsKit.Pages;
+using Volo.CmsKit.Public.Pages;
+
+namespace Volo.CmsKit.Public.Web;
+
+public class DefaultHomePageMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly IPagePublicAppService _pagePublicAppService;
+
+    public DefaultHomePageMiddleware(RequestDelegate next, IPagePublicAppService pagePublicAppService)
+    {
+        _next = next;
+        _pagePublicAppService = pagePublicAppService;
+    }
+    public async Task InvokeAsync(HttpContext httpContext)
+    {
+        var page = await _pagePublicAppService.FindDefaultHomePageAsync();
```

Comment:
Making a database query for each request is not acceptable for the final version of this feature. We should consider using caching here.

Reply:
`FindDefaultHomePageAsync` already uses caching

Reply:
https://github.com/abpframework/abp/pull/13635/files#diff-b0973506765b84539d2a37f37e02274bc1d2f438b00810fad8e843ba99f2224dR42

Right